docs/proposal/20211207-delegated-storage.md

```diff
@@ -0,0 +1,554 @@
+      }
+    ```
+
+    **TODO:** How we can enforce such convention? Type composition?
+
+1. To install new Storage Backend, Cluster Admin has two options:
+
+    - use Capact Actions (e.g. `cap.interface.capactio.capact.hub.storage.helm-release.install`).
+    - Register a storage backend by creating such TypeInstance.
+
+    Regardless the option, at the end there is one TypeInstance produced:
+      
+      ```yaml
+      id: 3ef2e4ac-9070-4093-a3ce-142139fd4a16
+      metadata:
+        name: helm-storage
+      typeRef:
+        path: cap.type.helm.storage
+        revision: 0.1.0
+      latestResourceVersion:
+        metadata:
+          attributes:
+          - path: cap.core.attribute.hub.storage.backend # related to GrapHQL implementation
+            revision: 0.1.0
+        value:
+          url: "helm-release.default:50051"
+          additionalParametersSchema: {
+            "$id": "#/properties/additionalParametersSchema",
+            "type": "object",
+            "required": [
+              "name",
+              "namespace"
+            ],
+            "properties": {
+              "name": {
+                "$id": "#/properties/additionalParametersSchema/properties/name",
+                "type": "string"
+              },
+              "namespace": {
+                "$id": "#/properties/additionalParametersSchema/properties/namespace",
+                "type": "string"
+              }
+            },
+            "additionalProperties": false
+          }
+      backend: capact-redis # immutable - contains TypeInstance ID or unique alias
+        # if not provided, fallback to default one 
+      ```
+
+1. In fresh Capact installation, there is one TypeInstance already preregistered:
+
+    **TODO:** Would Redis be really the default storage? Probably PostgreSQL will be better, as we treat it as a "builtin" Backend as well (to store Attributes, relations, etc.)
+
+    ```yaml
+    id: a36ed738-dfe7-45ec-acd1-8e44e8db893b
+    name: capact-redis
+    typeRef:
+        path: cap.core.type.hub.storage.redis
+        revision: 0.1.0
+    latestResourceVersion:
+      metadata:
+        attributes:
+        - path: cap.core.attribute.default # if more such Typeinstances with default Attribute, select first one
+          revision: 0.1.0 
+        - path: cap.core.attribute.hub.storage.backend
+          revision: 0.1.0
+      value:
+        url: "storagebackend-handlers.capact-system:50051"
+        additionalParameters: null
+    backend: builtin # the storage option which stores already all other metadata. It could be the same Redis instance
+    ```
+
+    Default storage backend should have `additionalParameters` empty, or, optional.
+
+    **TODO:** How to enforce that default backend doesn't have any additionalParameters needed? Maybe we shouldn't enforce that at all, as there could be additionalParameters, but not required. We could enforce that if it is implemented on Policy level...
```

Comment:
we can check that, if someone wants to set the `cap.core.attribute.default` attribute, we check the `additionalParameters` if there are no required fields, if yes, we reject that TI. 

We already have code for checking that:
https://github.com/capactio/capact/blob/3b13deecade8f967ed8d5dca832a7879a4baec8a/pkg/sdk/validation/interfaceio/validator.go#L174-L192

Reply:
I would rather avoid such additional logic for a given Attribute, but you are probably right, we need to do that 😞 I was trying to figure out some generic, universal way to do it, but maybe it's good to have it fixed for now.